lib/Futures/Utilities.h

```diff
@@ -158,6 +162,35 @@ auto collectAll(Collection&& c) -> decltype(collectAll(std::begin(c), std::end(c
   return collectAll(std::begin(c), std::end(c));
 }
 
+template <class Collection, class Rep, class Period>
+typename Collection::iterator waitForAny(Collection& c,
+                                         std::shared_ptr<std::condition_variable> cv,
```

Comment:
wait a sec, I think you should have a look at https://github.com/facebook/folly/blob/master/folly/futures/Future-inl.h#L1605

Reply:
@graetzer Yeah, I thought about doing something like that, but as far as I can tell, that approach then essentially discards all the other futures besides the one collected. Or at the very least, it modifies them in some way that I don't quite understand. So it's not clear to me that you could do something like:
```
std::vector<futures::Future<T>> collection;
...
while (!collection.empty()) {
  auto it = futures::waitForAny(collection);
  ...
  collection.erase(it);
}
```
The above outlines the semantics I need in this particular use case. I need to wait for all the futures eventually, but want to react to them as they come in (and possibly generate more in the process), rather than add a synchronization barrier like `collectAll`. Does that make sense?